### DISCUSSION TITLE
What `TimeDeltaSensor` actually do?

### DISCUSSION THREAD

**Discussion by makrushin-evgenii** [apache/airflow#35184] [Q&A]
I want the task to start some time after the previous one is completed. Find `TimeDeltaSensor` in the docs:
> Use the [TimeDeltaSensor](https://airflow.apache.org/docs/apache-airflow/stable/_api/airflow/sensors/time_delta/index.html#airflow.sensors.time_delta.TimeDeltaSensor) to end sensing after specific time.

I understand that `TimeDeltaSensor` is like `sleep()` method and fits perfectly. But his `poke` method looks like he waits until `delta` time after `data_interval_end`:
```
def poke(self, context: Context):
        target_dttm = context["data_interval_end"]
        target_dttm += self.delta
        self.log.info("Checking if the time (%s) has come", target_dttm)
        return timezone.utcnow() > target_dttm
```

If this is true, how to suspend DAG proper way?

**Comment by makrushin-evgenii** [comment]
I can handle it with `TimeSensor` like this. But I guess that it is common problem and has more elegant solution
```
@task 
def first_task():
  return pendulum.now()

first_task_finished_at = first_task()
TimeSensor(target_time=first_task_finished_at.add(hours=1))
```

**Reply by makrushin-evgenii** [reply]
Looks like I'm not the first one confused with `TimeDeltaSensor`: [1](https://stackoverflow.com/a/77116016/5516822), [2](https://stackoverflow.com/questions/59757151/airflow-implementing-wait-sleep-task-efficiently?noredirect=1&lq=1). May I fix it by retaining `now` at `__init()__` method?

```
def __init__(self, *, delta, **kwargs):
        super().__init__(**kwargs)
        self.init_time = pendulum.now()
        self.delta = delta

def poke(self, context: Context):
        self.log.info("Checking if the time (%s) has come", target_dttm)
        return timezone.utcnow() > self.init_time  + self.delta
```

It may cause backward compatibility problem for developers, who relied on the current behavior. But I can't find an explanation for it. When is it useful to wait after `data_interval_end`?

**Reply by Taragolis** [reply]
I can't tell you about the main idea of this sensor. However it behaviour at that moment predictable and mostly deterministic.

> self.init_time = pendulum.now()

This would not work well in case of:
- `mode = "reschedule"` each reschedule init_time would changed to current time, so most possible it would never finish
- On retries

**Reply by makrushin-evgenii** [reply]
> each reschedule init_time would changed to current time, so most possible it would never finish

Got it, thank you. I would use `TimeSensor` then. Or is there a way to get previous task finish time to use in sensor?
